### PR TITLE
export non-ambient types in ERTP

### PR DESCRIPTION
**Pull request by turadg**
refs: #6343

## Description

ERTP part of https://github.com/Agoric/agoric-sdk/issues/6343

Review by commit suggested.

### Security Considerations

n/a

### Documentation Considerations

Improves API documentation

### Testing Considerations

CI

### CODE REVIEW

File: packages/vats/src/core/types.js [hunk 1]
```diff
@@ -157,11 +157,11 @@
  * @typedef {{
  *   issuer: {
  *     produce: Record<WellKnownName['issuer'], Producer<Issuer>>,
- *     consume: Record<WellKnownName['issuer'], Promise<Issuer>>,
+ *     consume: Record<WellKnownName['issuer'], Promise<Issuer>> & { BLD: Promise<Issuer<'nat'>>, IST: Promise<Issuer<'nat'>> },
```
Comment (turadg) [inline packages/vats/src/core/types.js:160]: PTAL @dckc

File: packages/zoe/src/contractFacet/types.js [hunk 2]
```diff
@@ -81,7 +81,7 @@
  *
  * @param {ERef<Issuer>} issuerP Promise for issuer
  * @param {Keyword} keyword Keyword for added issuer
- * @returns {Promise<IssuerRecord>} Issuer is added and ready
+ * @returns {Promise<IssuerRecord<K>>} Issuer is added and ready
```
Comment (turadg) [inline packages/zoe/src/contractFacet/types.js:84]: `any` for keyword records since there's no way to know the actual type. We could keep `AssetKind` as the parameter, but that kind of means "all four of these" instead of "dunno"

File: packages/vats/src/core/types.js (hunk 1 above)
Comment (dckc) [reply packages/vats/src/core/types.js:160]: hm. yes, I guess that's reasonable.

File: packages/zoe/src/contractFacet/types.js (hunk 2 above)
Comment (dckc) [reply packages/zoe/src/contractFacet/types.js:84]: yeah; keyword records and static types don't get along well.

File: packages/zoe/test/unitTests/contracts/test-coveredCall.js [hunk 3]
```diff
@@ -1000,9 +1001,13 @@ test('zoe - coveredCall non-fungible', async t => {
     `The option was exercised. Please collect the assets in your payout.`,
   );
 
+  /** @type {Payment<any>} */
```
Comment (dckc) [inline packages/zoe/test/unitTests/contracts/test-coveredCall.js:1004]: Why is this needed?

File: packages/inter-protocol/src/feeDistributor.js [hunk 4]
```diff
@@ -171,9 +171,13 @@ export const makeFeeDistributor = (feeIssuer, terms) => {
...
 
-  /** @param {Payment} payment */
+  /** @param {Payment<'nat'>} payment */
   const distributeFees = payment =>
-    sharePayment(payment, feeIssuer, shareConfig);
+    sharePayment(
+      payment,
+      /** @type {Issuer<'nat'>} */ (feeIssuer),
+      shareConfig,
+    );
```
Comment (dckc) [inline packages/inter-protocol/src/feeDistributor.js]: this looks a little fiddly... help me understand why/if it's necessary?

File: packages/inter-protocol/src/my-lien.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ const XLien = /** @type { const } */ ({
  */
 export const makeStakeReporter = (bridgeManager, brand, denom = 'ubld') => {
   const { make: makeAmt } = AmountMath;
-  /** @param {string} numeral */
+  /** @type {(numeral: string) => Amount<'nat'>} */
```
Comment (dckc) [inline packages/inter-protocol/src/my-lien.js]: I gather this isn't needed.

File: packages/inter-protocol/test/test-voPool.js [hunk 6]
```diff
@@ -62,7 +62,7 @@ const voPoolTest = async (t, mutation, postTest) => {
 
   const { brand: secondaryBrand, issuer: secondaryI } =
     makeIssuerKit('secondary');
-  const quoteIssuerKit = makeIssuerKit('Quotes');
+  const quoteIssuerKit = makeIssuerKit('Quotes', 'set');
```
Comment (dckc) [inline packages/inter-protocol/test/test-voPool.js:65]: nifty... bug fix.

**Review by dckc — APPROVED** [review]
Looks like while scrubbing a bunch of stuff, the types were refined too. In some cases, it might add some burden/friction... but inter-protocol should be somewhat stable now, so that's probably worthwhile.
